pilot/pkg/proxy/envoy/v2/cds.go

```diff
@@ -78,7 +78,29 @@ func (s *DiscoveryServer) generateRawClusters(node *model.Proxy, push *model.Pus
 		return nil, err
 	}
 
+	if sdsTokenPath, found := node.Metadata[model.NodeMetadataSdsTokenPath]; found && len(sdsTokenPath) > 0 {
+		// If SDS_TOKEN_PATh is in the node metadata, make a copy of rawClusters so that
+		// the path of SDS token will be applied to the copied clusters.
+		clusters := make([]*xdsapi.Cluster, 0)
```

Comment:
As I mentioned, push context caches are not supposed to be updated on the fly - this is quite risky. 

If it must be done - it should have extensive tests to verify it doesn't introduce race conditions (since multiple goroutines may hit this).

Please add at least some test for this - it's performance-impacting and may result in crashes, we can't get it in 1.1 this late without any relevant tests.

Reply:
The path change is only applied to the copied clusters, which is local so it should not introduce race conditions.